cocos/root.ts

```diff
@@ -489,46 +489,14 @@ export class Root {
             this._frameCount = 0;
             this._fpsTime = 0.0;
         }
-        for (let i = 0; i < this._scenes.length; ++i) {
-            this._scenes[i].removeBatches();
-        }
-
-        const windows = this._windows;
-        const cameraList = this._cameraList;
-        cameraList.length = 0;
-
-        for (let i = 0; i < windows.length; i++) {
-            const window = windows[i];
-            window.extractRenderCameras(cameraList);
-        }
-
-        if (this._pipeline && cameraList.length > 0) {
-            this._device.acquire([deviceManager.swapchain]);
-            const scenes = this._scenes;
-            const stamp = director.getTotalFrames();
-
-            if (this._batcher) {
-                this._batcher.update();
-                this._batcher.uploadBuffers();
-            }
-
-            for (let i = 0; i < scenes.length; i++) {
-                scenes[i].update(stamp);
-            }
 
-            director.emit(Director.EVENT_BEFORE_COMMIT);
-            cameraList.sort((a: Camera, b: Camera) => a.priority - b.priority);
-
-            for (let i = 0; i < cameraList.length; ++i) {
-                cameraList[i].geometryRenderer?.update();
-            }
-            director.emit(Director.EVENT_BEFORE_RENDER);
-            this._pipeline.render(cameraList);
-            director.emit(Director.EVENT_AFTER_RENDER);
-            this._device.present();
+        if (globalThis.__globalXR.isWebXR) {
+            this._doWebXRFrameMove();
+        } else {
+            this._frameMoveBegin();
+            this._frameMoveProcess();
+            this._frameMoveEnd();
```

Comment:
Invoke functions instead may have lower performance, it is better to do performance test.

Reply:
The dowebxr process requires logic to be interspersed before and after these three functions. If the functions are not split, many judgments need to be added and interspersed within the frameMove. The readability of the code is not high. And this is also the unified code with the native, which is the three splitting functions.

Reply:
Root.cpp void Root::frameMove(float deltaTime, int32_t totalFrames) { // NOLINT
    CCObject::deferredDestroy();

    _frameTime = deltaTime;

    ++_frameCount;
    _cumulativeTime += deltaTime;
    _fpsTime += deltaTime;
    if (_fpsTime > 1.0F) {
        _fps = _frameCount;
        _frameCount = 0;
        _fpsTime = 0.0;
    }

    if (_xr) {
        doXRFrameMove(totalFrames);
    } else {
        frameMoveBegin();
        frameMoveProcess(true, totalFrames);
        frameMoveEnd();
    }
}